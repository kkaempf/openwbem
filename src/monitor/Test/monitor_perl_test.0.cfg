unpriv_user { owprovdr }

read_dir {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/
  /dir/doesnt/exist/
}

rename_from {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename1
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2
  /dir/doesnt/exist/rename
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file
}

rename_to {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
  /dir/doesnt/exist/rename-to
}

unlink {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-exists
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-doesnt-exist
  /dir/doesnt/exist/unlink
}

open_r {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/bar
}

open_rw {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-*
}

open_w {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-*
}

open_a {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-*
}

user_exec {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_user @ root
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_user @ owprovdr
}

monitored_exec {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_mon @ mpt_mon.cfg
}
