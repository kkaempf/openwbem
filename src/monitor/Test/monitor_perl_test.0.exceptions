PrivilegeManagerException   PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/
PrivilegeManagerException   PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/
