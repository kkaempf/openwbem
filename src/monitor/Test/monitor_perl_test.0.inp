# These test cases where there are no protocol errors

readDirectory /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/ keep_special
readDirectory /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/ omit_special
readDirectory /dir/doesnt/exist/ keep_special
readDirectory /dir/doesnt/exist/ omit_special

rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename1 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2 /dir/doesnt/exist/rename-to
rename /dir/doesnt/exist/rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

remove_file /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-exists
remove_file /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-doesnt-exist
remove_file /dir/doesnt/exist/unlink

open < 0 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/bar
open > 604 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-exists
  piffle-prog
open > 606 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-create
  hack-phlegm
open > 604 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-trunc-exists
  enigmatic
open > 604 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-trunc-create
  wizard-of-oz
open >> 700 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-exists
  send-in-the-clowns
open >> 500 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-create
  horatio-at-the-bridge
open +< 404 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-exists
  xxx
open +< 400 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-nonesuch
  foo-bar-baz
open +> 606 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-trunc-exists
  yyy
open +> 606 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-trunc-create
  This-is-the-song-that-never-ends
open +>> 402 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-exists-ate
  xxx
userSpawn /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_user a+b+c PATH=/bin:/usr/bin+FOO=bar root
userSpawn /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_user d PATH=/bin:/usr/bin+HIP=hop owprovdr
monitoredSpawn /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_mon mpt_mon.cfg /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir+a+b+c PATH=/bin:/usr/bin+FOO=bar
