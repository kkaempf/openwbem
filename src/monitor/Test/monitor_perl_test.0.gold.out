User: owprovdr

readDirectory
  .
  ..
  app-exists
  bar
  foo
  in-out-exists
  in-out-exists-ate
  in-out-trunc-exists
  link-exists
  out-exists
  out-trunc-exists
  rename1
  rename2
  these-are-the-link-contents
  unlink-exists

readDirectory
  app-exists
  bar
  foo
  in-out-exists
  in-out-exists-ate
  in-out-trunc-exists
  link-exists
  out-exists
  out-trunc-exists
  rename1
  rename2
  these-are-the-link-contents
  unlink-exists

readDirectory
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/

readDirectory
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/

rename

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2 to /dir/doesnt/exist/rename-to

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /dir/doesnt/exist/rename to /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file to /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

unlink
1

unlink
0

unlink
0

open
  Hokey smokes, Bullwinkle!


open

open

open

open

open

open

open
  xxxtle Bo Peep has lost her sheep


open
Privman::open_handle failed

open
  yyy

open
  This-is-the-song-that-never-ends

open
  Little Bo Peep has lost her sheep
xxx

userSpawn
report_process...
running: 1
--- output stream ---
uid == euid: 1
User: root
This is the input sent to mpt_user
argv: /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_user a b c
env: PATH=/bin:/usr/bin FOO=bar
--- error stream ---
--- termination status ---
running: 0
terminated: 1
terminated_successfully: 1
exit_terminated: 1
exit_status: 0
signal_terminated: 0
stopped: 0

userSpawn
report_process...
running: 1
--- output stream ---
uid == euid: 1
User: owprovdr
This is the input sent to mpt_user
argv: /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_user d
env: PATH=/bin:/usr/bin HIP=hop
--- error stream ---
--- termination status ---
running: 0
terminated: 1
terminated_successfully: 1
exit_terminated: 1
exit_status: 0
signal_terminated: 0
stopped: 0

monitoredSpawn
report_process...
running: 1
--- output stream ---
uid == euid: 1
User: owprovdr
This is the input sent to mpt_user
argv: /montest-689acb0e1ec89f45-7085a2a1780f5f42/safebin/mpt_mon /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir a b c
env: PATH=/bin:/usr/bin FOO=bar
read_dir:
  app-create
  app-exists
  bar
  foo
  in-out-exists
  in-out-exists-ate
  in-out-trunc-create
  in-out-trunc-exists
  link-exists
  out-create
  out-exists
  out-trunc-create
  out-trunc-exists
  rename-to
  rename2
  these-are-the-link-contents
--- error stream ---
--- termination status ---
running: 0
terminated: 1
terminated_successfully: 1
exit_terminated: 1
exit_status: 0
signal_terminated: 0
stopped: 0
